app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt

```diff
@@ -387,6 +388,22 @@ class ProfileEditActivityTest {
     }
   }
 
+  @Test
+  fun testProfileEditActivity_clickProfileDeletionButton_configurationChange_dialogPersists() {
+    ActivityScenario.launch<ProfileEditActivity>(
+      ProfileEditActivity.createProfileEditActivity(
+        context = context,
+        profileId = 1
+      )
+    ).use {
+      onView(withId(R.id.profile_delete_button)).perform(click())
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withText(R.string.profile_edit_delete_dialog_positive)).inRoot(isDialog()).check(
```

Comment:
Macros will help here - https://github.com/oppia/oppia-android/wiki/Ktlint-Guide

Reply:
> Thanks @prudhvir3ddy
> just a nit change
> 
> I had run the test in espresso, but the orientation changing is not visible on the screen, could you confirm if you are seeing the same or are you able to see the screen rotation?
> 
> Also, please check, the test should fail without the implementation, I guess, if we don't have your implementation, this test case will still pass.
> 
> orientation change is working fine for all other test cases in the same file, just re confirmed.

@anandwana001 thanks for the review.

you are not able to see the orientation because the check is happening too fast. replace isDialog check with some text matching check then you can visually see it. 

regarding failure case. yes test case is failing if I got no implementation but the problem is it's waiting for too much time to dialog to appear. 

should I write a wait time for around 2 seconds and fail the test case ?

Reply:
check `testProfileEditActivity_configurationChange_startActivityWithUserProfile_clickProfileDeletionButton_clickDelete_checkReturnsToProfileListActivity`, try to run this test in espresso, you will see the orientation change. 

For delay we use `TestCoroutineDispatchers`  - https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#tips-to-run-test-cases-in-both-espresso-and-robolectric

I think there is something we are missing, changing orientation when dialog is displayed on the screen.

Reply:
@prudhvir3ddy @rt4914 How about if we have a different implementation for dialog, just like `AdminSettingsDialogFragment` or `ExitProfileDialogFragment` or any other `*DialogFragment`?

What are your thoughts on this?

Reply:
@anandwana001 yes that's the ideal solution but I was worried I will generate a lot of code. 

if you are okay with that I'll go with that solution

Reply:
we can also have a common alert dialog throughout the app and pass what we need

Reply:
@anandwana001 I am not able able to understand you question.

Reply:
@rt4914 currently, the dialog implementation is inside the `ProfileEditActivityPresenter` so should we do the implementation the way we are doing in `AdminSettingsDialogFragment ` or `ExitProfileDialogFragment ` or for any other `*DialogFragment `

Reply:
@anandwana001 Both are correct approaches at their own places.If you notice `ExitProfileDialogFragment` and `AdminSettingsDialogFragment ` both are getting used at multiple places and therefore we have created this dialog fragment. But in this case `ProfileEditActivityPresenter ` it is getting used at single place only and therefore keeping it in presenter makes sense. 

WDYT?

Reply:
correct will keep it as it is.